### ISSUE TITLE
Shorts Break Feed

### ISSUE THREAD

**Issue by Travisplo** [OsaSoft/youtube-better-subscriptions#105]
When a short is loaded in the Subscription feed, it stops the extension from affecting the videos after the short. Shorts have a different URL than normal videos, that might be the cause? Using the Hide function hidden under the vertical . . . that pops up when hovering over a video and then reloading fixes it, but it would be ideal if Shorts didn't break them at all.

**Comment by HidyHo** [comment]
I'm also seeing similar weird behavior in Sub feed starting today.

**Comment by treverbrice** [comment]
It seems like the program goes through each video in order and breaks when it sees the first shorts video. I've been able to work around this myself my clicking on the 3 dots by the video title and selecting "hide" for all shorts content.

Maybe changing how it parses the videoID would fix it? My Javascript is way too rusty to figure it out.

**Comment by HidyHo** [comment]
Definitely appreciate OP's accurate analysis.  Getting rid of shorts (manually hiding them) first and then refreshing the subscriptions fixes all the following videos (until you hit the next short).

FYI, I had to hide even previously viewed shorts manually (damn, there were a lot of shorts in my feed), in order to have the following videos to function as expected.

**Comment by Travisplo** [comment]
> ALso, "shorts" video has new red "S" icon on the lower right corner of the video now. That's how I have been identifying shorts.

I've been identifying shorts by seeing where Watched videos start showing up in the feed. XP

**Comment by Urhas** [comment]
Issue looks to be down to the getVideoIdFromUrl function not being able to handle the new Youtube shorts URLs. I have submitted a fix to the devop branch

Line:106 - Subs.js (return url.split("=")[1].split("&")[0];)

**Comment by Drachlox** [comment]
Looking forward to a solution. 

I wish i could just remove shorts alltogether without adding another addon, but of course youtube have not added any option to disable YTshorts..

**Comment by Spurlos** [comment]
Yep, due to shorts this extension is currently broken from in its current state. Waiting patiently for an update from developer

**Comment by OsaSoft** [comment]
Strange. Its working correctly even with #shorts in my feed, both in FF and in Chrome. Might be some part of a YT update that has not rolled out to all users yet.
Can someone tell me what the new shorts urls on the videos are? For me, shorts in the subs feed still show a regular video url, eg. https://www.youtube.com/watch?v=iNoBdQN9UkQ

**Comment by Urhas** [comment]
Yea seems to be slowly rolling out. An example of the new shorts url is https://m.youtube.com/shorts/mvIue5q1Mg0

i submitted pull request 106 as a quick fix to handle these. Locally it has been working but unsure if theres a more graceful way to do it.

**Comment by Drachlox** [comment]
> Can someone tell me what the new shorts urls on the videos are? 

https://www.youtube.com/shorts/nlQTdXvBDjo

**Comment by OsaSoft** [comment]
@Drachlox ugh, that sounds terrible...

**Comment by OsaSoft** [comment]
Ok! Version 0.17.0 which includes this, the feature in #100 and a few other fixes is uploaded to Mozilla and Chrome and should be going live as soon as they are approved!
